include/caffe/net.hpp

```diff
@@ -192,7 +192,8 @@ class Net {
   /// @brief Append a new bottom blob to the net.
   int AppendBottom(const NetParameter& param, const int layer_id,
                    const int bottom_id, set<string>* available_blobs,
-                   map<string, int>* blob_name_to_idx);
+                   map<string, int>* blob_name_to_idx,
+                   bool skip_propagate = false);
```

Comment:
I can change it if you want, but in this way [this check](https://github.com/mtamburrano/caffe/blob/fe4dad53e65574573614f600d50aad06c96b8739/src/caffe/net.cpp#L83-L88) needs to be done in the `AppendBottom` function, that it is called `Bottom.size()` times.
I know that this is not a computational relevant part, but don't you think that will be a little bit dirty?